core-ui/src/shared/components/ResourceRef/ExternalResourceRef.js

```diff
@@ -54,68 +54,52 @@ export function ExternalResourceRef({
       title={title}
       tooltipContent={tooltipContent}
       actions={actions}
-      className={className}
+      className={classnames('external-resource-ref', className)}
       defaultOpen={open}
       required={required}
     >
-      <div className={classnames('fd-row form-field')}>
-        <div className="fd-col fd-col-md--4 form-field__label">
-          <Label
-            required={required}
-            tooltipContent={t('common.tooltips.secret-ref-namespace')}
-          >
-            {labelPrefix + ' '}
-            {t('common.labels.namespace')}
-          </Label>
-        </div>
-        <div className="fd-col fd-col-md--7">
-          <Dropdown
-            id="external-resource-dropdown"
+      <ResourceForm.FormField
+        required={required}
+        label={t('common.labels.resource-namespace', { resource: labelPrefix })}
+        tooltipContent={t('common.tooltips.secret-ref-namespace')}
+        input={() => (
+          <ComboboxInput
             compact
-            fullWidth
+            showAllEntries
+            searchFullString
             options={namespacesOptions}
             placeholder={t('common.placeholders.secret-ref-namespace')}
-            selectedKey={resourceRef.namespace}
-            onSelect={(e, selected) => {
-              onChange({
-                ...resourceRef,
+            value={value?.namespace}
+            onSelect={e => {
+              setValue({
                 name: undefined,
-                namespace: selected.key,
+                namespace: e.target.value,
               });
             }}
-            i18n={i18n}
           />
-        </div>
-      </div>
-      <div className={classnames('fd-row form-field')}>
-        <div className="fd-col fd-col-md--4 form-field__label">
-          <Label
-            required={required}
-            tooltipContent={t('common.tooltips.secret-ref-name')}
-          >
-            {labelPrefix + ' '}
-            {t('common.lowercase.name')}
-          </Label>
-        </div>
-        <div className="fd-col fd-col-md--7">
-          <Dropdown
-            id="external-resource-dropdown"
+        )}
+      />
+      <ResourceForm.FormField
+        required={required}
+        label={t('common.labels.resource-name', { resource: labelPrefix })}
+        tooltipContent={t('common.tooltips.secret-ref-name')}
+        input={() => (
+          <ComboboxInput
```

Comment:
If I choose the secret name first, and the namespace is not set, I have to choose the namespace manually. So, I can set a namespace that doesn't have this secret. Maybe we could either detect and choose a namespace based on a chosen secret or disable the secret dropdown until we choose a namespace?